docs/rest-api.raml

```diff
@@ -0,0 +1,50 @@
+swagger: "2.0"
+info:
+    title: "Ginger REST API documentation"
+    version: 0.1
+
+basePath: "/data"
+
+paths:
+    /resources:
+        get:
+            parameters:
+                - in: query
+                  name: cat
+                  schema:
+                      type: string
+                  required: false
+                  description: A category name
+                - in: query
+                  name: hasobject
+                  schema:
+                      type: string
+                  required: false
+                  description: A predicate name (of an outgoing edge)
+                - in: query
+                  name: hassubject
+                  schema:
+                      type: string
+                  required: false
+                  description: A predicate name (of an incoming edge)
+    /resources/{id}:
+        get:
+            parameters:
+                - in: path
+                  schema:
+                      type: integer
+                  required: true
+    /resources/{id}/incoming-edges:
+        get:
+            parameters:
+                - in: path
+                  schema:
+                      type: integer
+                  required: true
+    /resources/path/{path}:
+        get:
+            parameters:
+                - in: path
+                  schema:
+                      type: integer
+                  required: true
```

Comment:
How do these definitions compare to JSON-LD, the format that we want to use for our API? 

> As a result, different sites have chosen their own proprietary representations for interacting with their sites, sometimes described using frameworks such as [swagger] which imply a particular URI composition for interacting with their services. This practice leads to vendor-specific semantic silos, where the meaning of a particular JSON document makes sense only by programming directly to the API documentation for a given service.

https://json-ld.org/spec/latest/json-ld-api-best-practices/